### PR TITLE
WebGLRenderer: Release cache at the end of .render().

### PR DESCRIPTION
**Pull request by takahirox**
I think it's better to release `_currentCamera` at the end of `WebGLRenderer`'s `.render()`. Unless doing this, disposed `camera` resource won't be released until next `.render()` call because `renderer` keeps a reference to `camera`.

### CODE REVIEW

File: src/renderers/WebGLRenderer.js [hunk 1]
```diff
@@ -1035,7 +1035,6 @@ function WebGLRenderer( parameters ) {
 		_currentGeometryProgram.program = null;
 		_currentGeometryProgram.wireframe = false;
 		_currentMaterialId = - 1;
-		_currentCamera = null;
```
[the remark this answers is not in the document]
Comment (takahirox) [reply src/renderers/WebGLRenderer.js]: Probably it's ok. Do you think we should move the block for the readability tho they have just primitives so won't cause memory leak issues?

[the remark this answers is not in the document]
Comment (Mugen87) [reply src/renderers/WebGLRenderer.js]: > Do you think we should move the block for the readability

I would say yes. Then the reset happens at a single place.

[the remark this answers is not in the document]
Comment (takahirox) [reply src/renderers/WebGLRenderer.js]: OK, updated.

**Review by Mugen87 — APPROVED** [review]

**Comment by mrdoob** [comment]
Say that we're rendering a scene with just one object and one material...
Wouldn't be doing unnecessary binding calls when rendering again?

**Comment by takahirox** [comment]
What does "binding calls" mean by in this context?

**Comment by takahirox** [comment]
Ah, binding buffer, isn't it?

Even in the current impl it binds every frame for a scene with one object and one material because it releases the cache at the beginning of `.render()`. So this PR change nothing about that.

**Comment by Mugen87** [comment]
Friendly bump 😇. I think it makes still sense to merge this PR.

**Comment by mrdoob** [comment]
> Even in the current impl it binds every frame for a scene with one object and one material because it releases the cache at the beginning of `.render()`. So this PR change nothing about binding calls.

I never understood that explanation. @takahirox do you mind elaborating?

**Comment by takahirox** [comment]
That's the answer to your question.

> Say that we're rendering a scene with just one object and one material...
Wouldn't be doing unnecessary binding calls when rendering again?

My understanding is you think

* Unnecessary binding calls can happen by clearing caches because cache is used for avoiding unnecessary binding calls by remembering the latest WebGL state
* Especially in the case we're rendering a scene with just one object and one material, WebGL state shouldn't change in each `renderer.render()` call so if we clear the caches at the end of `renderer.render()` unnecessary binding calls happen in the second or later `renderer.render()` call

correct?

I don't think this PR change the situation. Because even in the current implementation renderer clears the caches at the beginning of `renderer.render()`. After [this line](https://github.com/mrdoob/three.js/blob/r113/src/renderers/WebGLRenderer.js#L1152) in the renderer, the caches are cleared regardless of this PR. (If we want to avoid unnecessary binding calls in second or later frame `renderer.render()` call in the scenario you mentioned, we shouldn't clear the caches each frame.)

My original purpose is letting renderer release `_currentCamera` cache when `renderer.render()` finishes. Currently renderer keeps the `_currentCamera` cache until next `renderer.render()` call so even if user disposes the camera used in previous `renderer.render()` call the camera resource won't be released until next `renderer.render()` call because renderer keeps the reference to the camera until then. 

Let me know if I misunderstand anything.

**Comment by mrdoob** [comment]
> I don't think this PR change the situation. 

I think it would be good to support that hipothesis  comparing Spector.js traces with/without this change.

**Comment by takahirox** [comment]
Confirming sounds a nice idea.

Anyone knows if there is "a scene with just one object and one material" example in our examples? (Sorry I'm too lazy to create an example!)

**Comment by Mugen87** [comment]
Try it with this one: https://threejs.org/examples/webgl_geometry_cube

**Comment by Mugen87** [comment]
@mrdoob This PR is about when a cache or cache-like variable should be resetted.

Similar like the reset of render states and lists, the mentioned code block should be moved at the end of `render()`. In this way, the cache variables do no hold references to objects after the method has been quitted.

**Comment by mrdoob** [comment]
@Mugen87 Is it safe to merge?

**Comment by mrdoob** [comment]
I guess there is only one way to know... 🤓

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**